src/nv_ingest/stages/storages/embedding_storage_stage.py

```diff
@@ -0,0 +1,148 @@
+
+    storage_options = {
+        "key":access_key,
+        "secret": secret_key,
+        "client_kwargs": {
+        "endpoint_url": _DEFAULT_READ_ADDRESS
+        }
+    }
+
+    meta = df["metadata"]
+    emb_df_list = []  
+    file_uuid = uuid.uuid4().hex
+    destination_file = f"{bucket_path}/{file_uuid}.parquet"
+    write_path = f"s3://{bucket_name}/{destination_file}"
+    for idx, row in df.iterrows():
+        uu_id = row["uuid"]
+        metadata = row["metadata"].copy()
+
+        metadata["source_metadata"]["source_location"] = write_path
+
+        if row["document_type"] == ContentTypeEnum.EMBEDDING:
+            logger.debug("Storing embedding data to Minio")
+            metadata["embedding_metadata"][
+                "uploaded_embedding_url"
+            ] = write_path
+        # TODO: validate metadata before putting it back in.
+        # cm = str(metadata["content_metadata"])
+        text = metadata["content"]
+        # source_meta = str(metadata["source_metadata"])
+        emb = metadata["embedding"]
+        if emb is not None:
+            df.at[idx, "metadata"] = metadata
+            emb_df_list.append([emb, text])
+    
+    emb_df = pd.DataFrame(emb_df_list, columns=["vector", "text"])
+    logger.error(f"exporting: {emb_df}")
+    emb_df.to_parquet(write_path, engine='pyarrow', storage_options=storage_options)
+
+    return df
+
+
+def _store_embeddings(df, task_props, validated_config, trace_info=None):
+    logger.error(f"in store embedding")
+    try:
+        content_types = {}
+        content_types[ContentTypeEnum.EMBEDDING] = True
+
+        params = task_props.get("extra_params", {})
+        params["content_types"] = content_types
+
+        df = upload_embeddings(df, params)
+        
+        return df
+    except Exception as e:
+        traceback.print_exc()
+        err_msg = f"Failed to store embeddings: {e}"
+        logger.error(err_msg)
+        raise
+
+def generate_embedding_storage_stage(
+    c: Config,
+    task: str = "store_embedding",
+    task_desc: str = "Store_embeddings_minio",
+    pe_count: int = 24,
+):
+    """
+    Helper function to generate a multiprocessing stage to perform pdf content extraction.
+
+    Parameters
+    ----------
+    c : Config
+        Morpheus global configuration object
+    embedding_storage_config : dict
+        Configuration parameters for embedding storage.
+    task : str
+        The task name to match for the stage worker function.
+    task_desc : str
+        A descriptor to be used in latency tracing.
+    pe_count : int
+        Integer for how many process engines to use for pdf content extraction.
+
+    Returns
+    -------
+    MultiProcessingBaseStage
+        A Morpheus stage with applied worker function.
+    """
+    validated_config = EmbeddingStorageModuleSchema()
+    _wrapped_process_fn = functools.partial(_store_embeddings, validated_config=validated_config)
+
+    return MultiProcessingBaseStage(
```

Comment:
Here is the multi processing conversion for the embedding storage task. @drobison00